cmd/pinsharesvc/process.go

```diff
@@ -0,0 +1,501 @@
+		logFile.Close()
+		return fmt.Errorf("failed to start PinShare: %w", err)
+	}
+
+	pm.logInfo(fmt.Sprintf("PinShare backend started with PID %d", pm.pinshareCmd.Process.Pid))
+
+	// Create exit channel and monitor process in background
+	pm.pinshareExited = make(chan struct{})
+	go pm.monitorProcess(ctx, pm.pinshareCmd, "PinShare", pm.pinshareExited)
+
+	return nil
+}
+
+// monitorProcess monitors a process and logs when it exits.
+// The exited channel is closed when the process exits, allowing Stop functions to wait.
+func (pm *ProcessManager) monitorProcess(ctx context.Context, cmd *exec.Cmd, name string, exited chan struct{}) {
+	defer close(exited)
+
+	err := cmd.Wait()
+
+	select {
+	case <-ctx.Done():
+		// Context cancelled, expected shutdown
+		pm.logInfo(fmt.Sprintf("%s process exited (shutdown requested)", name))
+	default:
+		// Unexpected exit
+		if err != nil {
+			pm.logError(fmt.Sprintf("%s process exited unexpectedly", name), err)
+		} else {
+			pm.logInfo(fmt.Sprintf("%s process exited", name))
+		}
+	}
+}
+
+// StopIPFS stops the IPFS daemon
+func (pm *ProcessManager) StopIPFS() error {
+	pm.processMu.Lock()
+	defer pm.processMu.Unlock()
+
+	if pm.ipfsCmd == nil || pm.ipfsCmd.Process == nil {
+		return nil
+	}
+
+	pm.logInfo("Stopping IPFS daemon...")
+	pid := pm.ipfsCmd.Process.Pid
+
+	// On Windows, use taskkill to properly terminate the process tree
+	// os.Interrupt doesn't work reliably for processes created with CREATE_NEW_PROCESS_GROUP
+	killCmd := exec.Command("taskkill", "/F", "/T", "/PID", fmt.Sprintf("%d", pid))
+	if output, err := killCmd.CombinedOutput(); err != nil {
+		pm.logError(fmt.Sprintf("taskkill failed for IPFS (PID %d): %s", pid, string(output)), err)
+		// Fallback to process.Kill()
+		if err := pm.ipfsCmd.Process.Kill(); err != nil {
+			pm.logError("Failed to kill IPFS process", err)
+		}
+	}
+
+	// Wait for process to exit via the monitor goroutine (with timeout)
+	// This avoids calling Wait() twice which causes a race condition
+	if pm.ipfsExited != nil {
+		select {
+		case <-time.After(winservice.ProcessShutdownTimeout):
+			pm.logError("IPFS shutdown timeout", nil)
+		case <-pm.ipfsExited:
+			// Process exited, monitor goroutine has called Wait()
+		}
+	}
+
+	// Close log file
+	if pm.ipfsLogFile != nil {
+		pm.ipfsLogFile.Close()
+		pm.ipfsLogFile = nil
+	}
+
+	pm.ipfsCmd = nil
+	pm.logInfo("IPFS daemon stopped")
+	return nil
+}
+
+// StopPinShare stops the PinShare backend
+func (pm *ProcessManager) StopPinShare() error {
+	pm.processMu.Lock()
+	defer pm.processMu.Unlock()
+
+	if pm.pinshareCmd == nil || pm.pinshareCmd.Process == nil {
+		return nil
+	}
+
+	pm.logInfo("Stopping PinShare backend...")
+	pid := pm.pinshareCmd.Process.Pid
+
+	// On Windows, use taskkill to properly terminate the process tree
+	// os.Interrupt doesn't work reliably for processes created with CREATE_NEW_PROCESS_GROUP
+	killCmd := exec.Command("taskkill", "/F", "/T", "/PID", fmt.Sprintf("%d", pid))
```

Comment:
It seems like we're making similar calls to exec.Command to kill processes in multiple places. Can we factor out (or perhaps do we already have) a function which de-duplicates this logic for the most part?